### PR TITLE
Bh-231 Update documentation for upgrade from 4.0 to 5.0

### PR DESCRIPTION
**Pull request by rfauglas**
**Description**

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Technical Review and 2 GTM        | Todo
| English Review and 1 GTM          | Todo

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+Akeneo PIM composer.json
+----------------------------
+The root of your current installation dir is referred as $INSTALLATION_DIR.
+
+.. warning::
+    If you're on a local or on-premise environment, you need to change the APP_ENV to prod_onprem (see https://symfony.com/doc/current/configuration.html#selecting-the-active-environment)
+
+.. code:: bash
+    $ export APP_ENV=prod_onprem
```
Comment (rfauglas) [inline migrate_pim/upgrade/upgrade_from_40_to_50.rst:96]: has to be done later... Does't exist in 4.0...

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    $ cd $INSTALLATION_DIR
+    $ cp -R ./vendor/akeneo/pim-community-dev/upgrades/* ./upgrades/
+    $ cp -R ./vendor/akeneo/pim-enterprise-dev/upgrades/* ./upgrades/
+    $ php bin/console doctrine:migrations:version --add --all -q
```
Comment (rfauglas) [inline migrate_pim/upgrade/upgrade_from_40_to_50.rst:100]: Can we play it with php 7.4: tested with php 7.3...

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+    So there's no need to export and reimport data for this system.
+
+
+Updated System dependencies
+---------------------------
+
+Community Edition
+^^^^^^^^^^^^^^^^^
+TODO?
```
Comment (mr-prud) [inline migrate_pim/upgrade/upgrade_from_40_to_50.rst:38]:
```suggestion
You can download the composer.json file directly from the Github repository: https://raw.githubusercontent.com/akeneo/pim-community-standard/5.0/composer.json
```

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+
+Updated System dependencies
+---------------------------
+
+Community Edition
+^^^^^^^^^^^^^^^^^
+You can download the composer.json file directly from the Github repository: https://raw.githubusercontent.com/akeneo/pim-community-standard/5.0/composer.json
+
+Enterprise Edition
```
Comment (rfauglas) [inline migrate_pim/upgrade/upgrade_from_40_to_50.rst:37]: Suppressed: documented in sytem requirements
@mr-prud

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+
+Updated System dependencies
+---------------------------
+Check your system dependencies are in sync with :doc:`/install_pim/manual/system_requirements/system_requirements`
+Enterprise Edition
+^^^^^^^^^^^^^^^^^^
+
+.. warning::
+    The aspell dependencies are new.
```
Comment (mr-prud) [inline migrate_pim/upgrade/upgrade_from_40_to_50.rst:37]: Perhaps a warning about the need of the latest composer >=2 is important too.

File: migrate_pim/upgrade/upgrade_from_40_to_50.rst (hunk 4 above)
Comment (mr-prud) [reply migrate_pim/upgrade/upgrade_from_40_to_50.rst:33]: yes!

**Review by mr-prud — APPROVED** [review]

**Review by mr-prud — APPROVED** [review]
Only add the need of the latest composer , its more important than the aspell component

**Review by franckbrouard — APPROVED** [review]

**Pull request merged**